champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/BloodCompass.java

```diff
@@ -0,0 +1,267 @@
+package me.mykindos.betterpvp.champions.champions.skills.skills.warlock.passives;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.champions.champions.ChampionsManager;
+import me.mykindos.betterpvp.champions.champions.skills.Skill;
+import me.mykindos.betterpvp.champions.champions.skills.types.CooldownSkill;
+import me.mykindos.betterpvp.champions.champions.skills.types.ToggleSkill;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilBlock;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilPlayer;
+import org.bukkit.*;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.util.Vector;
+
+import java.util.*;
+
+@Singleton
+@BPvPListener
+public class BloodCompass extends Skill implements ToggleSkill, CooldownSkill {
+
+    private int numLines;
+    public double maxDistance;
+    public int effectDuration;
+    public int numPoints;
+    public double escapeRadius;
+    public double damage;
+    private final List<List<Location>> markers = new ArrayList<>();
+    private final Map<Integer, Player> lineToPlayerMap = new HashMap<>();
+    private Map<Integer, Integer> currentPoints;
+    private int taskId = -1;
+
+
+    @Inject
+    public BloodCompass(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Blood Compass";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "Drop Sword / Axe to activate",
+                "",
+                "Shoot out up to <val>" + (numLines + (level-1)) + "</val> tracking lines that will fly",
+                "towards the nearest enemies within <val>" + (maxDistance * level) +"</val> blocks",
+                "",
+                "Players hit with these blood lines will recieve <effect>Glowing</effect> for <stat>" + effectDuration + "</stat> seconds",
+                "and take <val>" + damage + ((level-1) * 2) +"</val> damage",
```

Comment:
Anything to do with level (and even things that don't have to do with level) should have a modifier per level. See https://github.com/Mykindos/BetterPvP/champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/Cleanse.java#L64. This makes it really easy to change how a level should interact with the skill.